name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/handlers/UploadToMyDividends24Handler.java

```diff
@@ -14,26 +15,46 @@
 import org.eclipse.e4.core.di.extensions.Preference;
 import org.eclipse.e4.ui.model.application.ui.basic.MPart;
 import org.eclipse.e4.ui.services.IServiceConstants;
+import org.eclipse.jface.dialogs.Dialog;
 import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.jface.layout.TableColumnLayout;
+import org.eclipse.jface.viewers.ArrayContentProvider;
+import org.eclipse.jface.viewers.ColumnWeightData;
 import org.eclipse.jface.viewers.LabelProvider;
+import org.eclipse.jface.viewers.StructuredSelection;
+import org.eclipse.jface.viewers.TableViewer;
+import org.eclipse.jface.viewers.TableViewerColumn;
 import org.eclipse.jface.window.Window;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.FillLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
 import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Label;
 import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Table;
 
+import name.abuchen.portfolio.model.Portfolio;
 import name.abuchen.portfolio.money.CurrencyConverter;
 import name.abuchen.portfolio.money.CurrencyConverterImpl;
 import name.abuchen.portfolio.online.impl.MyDividends24Uploader;
 import name.abuchen.portfolio.ui.Messages;
 import name.abuchen.portfolio.ui.PortfolioPlugin;
 import name.abuchen.portfolio.ui.UIConstants;
-import name.abuchen.portfolio.ui.dialogs.ListSelectionDialog;
 import name.abuchen.portfolio.ui.editor.ClientInput;
 import name.abuchen.portfolio.ui.jobs.AbstractClientJob;
 import name.abuchen.portfolio.ui.util.swt.ActiveShell;
+import name.abuchen.portfolio.ui.util.viewers.CopyPasteSupport;
 import name.abuchen.portfolio.util.Pair;
 
 public class UploadToMyDividends24Handler
 {
+    record Selection(String stringValue, Portfolio portfolio)
```

Comment:
Naming of "stringValue".

To be honest, it is super confusing to understand what the right identifier of a MY24 depot is.
The MyDividends24Uploader#getPortfolios method retrieves a list of pairs with the index and a name.
Reviewing the code, I think the index is never used later on.
Instead the name is eventually put into this stringValue property and used in the API.

You can see this also from the method signatures: #getPortoflios retrieves a list of (int, String), but #upload only takes a String.

My proposal therefore is:
* Refactor to the getPortfolios method to just return the list of names, i.e., drop the unnecessary index
* Rename this variable to something meaningful. I could imagine MY24Portfolio[Id] because it is about a "remote/external" portfolio. Below you use MyDividendsPortfolio which is also okay.